src/components/item/Item.js

```diff
@@ -1,85 +1,139 @@
-                            M
+                        id={id}
+                        onClick={() => setSize('m')}
+                    >
+                      M
                     </span>
                     <span 
-                        className={`cursor ${props.product.size.l > 0 ? 'font-weight-normal' : 'out-of-stock'} ${pickSizeToColor('l')}`} 
-                        style={{ marginRight: '1rem' }} 
-                        id={props.product.product_id}
-                        onClick={e => dispatch({ type: 'setCurrentSize', payload: { size: 'l', id: e.target.id } })}>
-                            L
+                        className={`
+                          cursor
+                          ${size === 'l' ? 'size' : ''}
+                          ${inventory.l > 0 ? 'font-weight-normal' : 'out-of-stock'}
+                        `} 
+                        style={{ marginRight: '1rem' }}
+                        id={id}
+                        onClick={() => setSize('l')}
+                    >
+                      L
                     </span>
                 </p>
-                <div className="card-text" style={{ marginBottom: '16px' }}>
+                <div
+                  className={`
+                    card-text
+                    ${formErrors.quantity ? 'invalid' : ''}
+                  `}
+                  style={{ marginBottom: '16px' }}
+                >
                     <span>Quantity: </span>
-                    <select value={numberOfItems.number} className="custom-select" id={props.product.product_id} onChange={e => setNumberOfItems({ number: e.target.value, id: e.target.id })}>
+                    <select
+                      value={quantity}
+                      className="custom-select"
+                      id={id}
+                      onChange={({ target: { value } }) => setQuantity(Number(value))}
+                    >
+                        <option value="0">0</option>
                         <option value="1">1</option>
                         <option value="2">2</option>
                         <option value="3">3</option>
+                        <option value="4">4</option>
+                        <option value="5">5</option>
+                        <option value="6">6</option>
+                        <option value="7">7</option>
                     </select>
                 </div>
-                <p className="card-text">Price: ${props.product.price}</p>
-                <button 
-                    className="btn btn-light" 
-                    id={props.product.product_id} 
-                    onClick={(e)=> {
-                        props.addToCart(e.target.id, numberOfItems);
-                        setNumberOfItems({ number: 1, id: null });
-                    }}
-                    >Add to <span role="img" aria-label="cart">🛒</span>
+                {formErrors.insufficientStock && (
```

Comment:
I was too lazy to add one more refactor to make form state global, but this is how you manage form state in react. Just declare your form state (typically done with other libraries anyway) and then add it to the UI. It's like...the opposite of jquery.